src/main/resources/compendium.yaml

```diff
@@ -0,0 +1,250 @@
+            characters
+          schema:
+            $ref: "#/components/schemas/ProtocolId"
+        - name: version
+          in: query
+          required: false
+          description: the version number of the requested protocol (must be greater than 0)
+          schema:
+            type: integer
+            minimum: 1
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Protocol"
+        "400":
+          description: Bad request
+        "404":
+          description: The protocol was not found for the requested id
+        "500":
+          description: Internal server error
+  "/protocol/{protocolId}/transformation":
+    get:
+      tags:
+        - protocol
+      summary: get a protocol by id and transform it into a different format
+      parameters:
+        - name: protocolId
+          in: path
+          required: true
+          description: the id of the protocol - can be a string of any alphanumeric
+            character, dash (-), or period (.) with a length less than or equal to 255
+            characters
+          schema:
+            $ref: "#/components/schemas/ProtocolId"
+        - name: target
+          in: query
+          required: true
+          description: the desired target protocol type of the transformation - must be one
+            of avro, protobuf, mu, openapi, or scala
+          schema:
+            $ref: "#/components/schemas/IdlName"
+        - name: version
+          in: query
+          required: false
+          description: the version number of the requested protocol - must be greater than 0
+          schema:
+            type: integer
+            minimum: 1
+      responses:
+        "200":
+          description: successful transformation from one stored protocol to another
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/FullProtocol"
+        "400":
+          description: Bad request
+        "404":
+          description: The protocol was not found for the requested id
+        "500":
+          description: Internal server error
+components:
+  schemas:
+    HealthResponse:
+      type: object
+      required:
+        - status
+        - serviceId
+        - version
+      properties:
+        status:
+          type: string
+          description: the status representing the health of the build
+        serviceId:
+          type: string
+          description: the name of the library artifact
+        version:
+          type: string
+          description: the library artifact version
+    IdlName:
+      type: string
+      description: the names of the set of protocols supported in compendium
+      enum:
+        - avro
+        - protobuf
+        - mu
+        - openapi
+        - scala
```

Comment:
As far as I know, right now, we are only supporting `avro` and `protobuf` idls to store in compendium (OpenAPI is still WIP).

@Yawolf is writing the docs about this right now. Could you confirm this?

```suggestion
```

Reply:
Nope, We can store any of the IDLs, but we can only translate:
Avro -> Mu
Protobuf -> Mu

Reply:
@Yawolf @pirita see https://github.com/higherkindness/compendium/blob/master/src/main/scala/higherkindness/compendium/core/ProtocolUtils.scala#L63-L87

Reply:
🤔 And why are we not allowing the others? I mean, for storing there should be no problem

Reply:
We should validate them before persisting them, and is still in progress

Reply:
Ah, right right... thank you

Reply:
Are you going to apply this one? :thinking: